src/chips/overflow/overflow_check.rs

```diff
@@ -39,14 +39,36 @@ impl<const MAX_BITS: u8, const ACC_COLS: usize> OverflowChip<MAX_BITS, ACC_COLS>
                     .map(|i: usize| meta.query_advice(decomposed_values[i], Rotation::cur()))
                     .collect::<Vec<_>>();
 
+                // multiplier by position of accumulator column
+                // e.g. for ACC_COLS = 3, multiplier = [2^(2*MAX_BITS), 2^MAX_BITS, 1]
+                let multiplier = |pos: usize| {
+                    let mut shift_chunk = Fp::one();
+                    for _ in 1..pos {
+                        shift_chunk *= Fp::from(1 << MAX_BITS);
+                    }
+                    Expression::Constant(shift_chunk)
+                };
+
+                // We are performing an important calculation here to check for overflow in finite field numbers.
+                // A single range table is utilized which applies `1 << 8` to decompose the columns for range checking.
+                //
+                // Consider the example where ACC_COLS = 3, the decomposed values would be represented as follows:
+                //
+                // |     | a_0 (value) | a_1  | a_2  | a_3  |
+                // |-----|-------------|------|------|------|
+                // |  x  | 0xffffff    | 0xff | 0xff | 0xff |
+                //
+                // Here, each column `a_n` represents a decomposed value.
+                // So, decomposed_value_sum would be calculated as a_0 * 2^16 + a_1 * 2^8 + a_2 * 1.
```

Comment:
Thanks for the refactoring. I think there's a typo here in the comment as I guess you meant 

```
a_1 * 2^16 + a_2 * 2^8 + a_3* 1.
```

Also, just to be sure, the decomposition in performed little endian, namely the most significant bits is in the first advice column?

Reply:
Thanks for checking it!

> Also, just to be sure, the decomposition in performed little endian, namely the most significant bits is in the first advice column?

Yes, `a_1` is MSB in the description. also, fixed at [477c0cd](https://github.com/summa-dev/summa-solvency/pull/33/commits/477c0cdc7e81bef6d3d77c89a12a59bc2307a398)